userops_ext.go

```diff
@@ -343,11 +346,7 @@ func (op *UserOperation) UnmarshalJSON(data []byte) error {
 	if intentJSON, ok := ExtractJSONFromField(aux.CallData); ok {
 		op.CallData = []byte(intentJSON)
 	} else {
-		var err error
-		op.CallData, err = hexutil.Decode(aux.CallData)
-		if err != nil {
-			return fmt.Errorf("invalid CallData: %w", err)
-		}
+		return fmt.Errorf("invalid CallData: data is not valid protobuf JSON")
```

Comment:
👍 , as per the function doc header, the function is only concerned with JSON unmarshalling.